src/components/UserSelector.tsx

```diff
@@ -1,16 +1,64 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import * as usersService from '../api/users';
+
+import { User } from '../types/User';
+import cn from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  setIsError: (error: boolean) => void;
+  setUserSelectedId: (userId: number) => void;
+  isActive: boolean;
+  setIsActive: (isActive: boolean) => void;
+  setPostsByUser: Dispatch<SetStateAction<Post[]>>;
+};
+export const UserSelector: React.FC<Props> = props => {
+  const {
+    setIsError,
+    setUserSelectedId,
+    isActive,
+    setIsActive,
+    setPostsByUser,
+  } = props;
+
+  const [users, setUsers] = useState<User[]>([]);
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    usersService
+      .getUsers()
+      .then(setUsers)
+      .catch(() => setIsError(true));
+  }, [setIsError]);
+
+  useEffect(() => {
+    setIsActive(false);
+  }, [setIsActive]);
```

Comment:
The `useEffect` hook here resets the `isActive` state to false whenever `setIsActive` changes. This might not be necessary and could lead to unexpected behavior if `setIsActive` is updated elsewhere. Consider whether this effect is needed.

Reply:
> The `useEffect` hook here resets the `isActive` state to false whenever `setIsActive` changes. This might not be necessary and could lead to unexpected behavior if `setIsActive` is updated elsewhere. Consider whether this effect is needed.

Good point

Reply:
> > The `useEffect` hook here resets the `isActive` state to false whenever `setIsActive` changes. This might not be necessary and could lead to unexpected behavior if `setIsActive` is updated elsewhere. Consider whether this effect is needed.
> 
> Good point

Not fixed